cli/cmd/plugin/ui/web/tanzu-ui/src/shared/components/FormInputSections/NodeInstanceTypeSection.tsx

```diff
@@ -0,0 +1,78 @@
+// React imports
+import React, { ChangeEvent } from 'react';
+// Library imports
+import { CdsControlMessage } from '@cds/react/forms';
+import { CdsIcon } from '@cds/react/icon';
+import * as yup from 'yup';
+import { FieldError } from 'react-hook-form';
+import { CdsRadio, CdsRadioGroup } from '@cds/react/radio';
+// App imports
+import './NodeInstanceTypeSection.scss';
+
+export interface NodeInstanceType {
+    id: string;
+    icon: string;
+    isSolidIcon?: boolean;
+    label: string;
+    description: string;
+}
+
+/**
+ * addNodeInstanceTypeValidation takes a "yup" schema object and returns a new "yup" schema object with the field added,
+ * and associated with a yup validation
+ * @param field - the name of the field for the node instance id
+ * @param yupObject - the yup schema object
+ */
+export function addNodeInstanceTypeValidation(field: string, yupObject: any): any {
+    return { ...yupObject, [field]: yup.string().nullable().required('Please select an instance type for your cluster nodes') };
+}
+
+function InstanceTypeInList(field: string, instance: NodeInstanceType, register: any, selectedInstanceId?: string) {
+    return (
+        <CdsRadio cds-layout="m:lg m-l:xl p-b:sm" key={instance.id + '-cds-radio'} data-testid="cds-radio">
+            <label>
+                {instance.label}
+                <CdsIcon
+                    shape={instance.icon}
+                    size="md"
+                    className={selectedInstanceId === instance.id ? 'node-icon selected' : 'node-icon'}
+                    solid={instance.isSolidIcon}
+                ></CdsIcon>
+                <div className="radio-message">{instance.description}</div>
+            </label>
+            <input type="radio" key={instance.id} value={instance.id} checked={selectedInstanceId === instance.id} readOnly />
+        </CdsRadio>
+    );
+}
+
+const DEFAULT_PROMPT = 'Select a node instance type';
```

Comment:
Should this be node profile instead of node instance type, since it sets the dev/prod plan along with underlying instance type?